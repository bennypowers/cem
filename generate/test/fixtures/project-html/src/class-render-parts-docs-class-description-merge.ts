/**
 * class description
 * @csspart summary - jsdoc part description
 */
@customElement('class-render-parts-docs-class-description-merge')
class ClassRenderPartsDocsClassDescriptionMerge extends LitElement {
  render() {
    return html`
      <!--
          summary: part summary
          description: |
            part description

            multiline **markdown**
      -->
      <div part="part">${childPart}</div>
      <!-- summary: summary part summary -->
      <div part="summary"></div>
      <!-- deprecated: true -->
      <div part="deprecated"></div>
      <!-- deprecated: deprecation reason -->
      <div part="reason"></div>
    `;
  }
}
