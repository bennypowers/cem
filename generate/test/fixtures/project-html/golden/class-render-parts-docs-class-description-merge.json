{
  "schemaVersion": "2.1.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/class-render-parts-docs-class-description-merge.js",
      "declarations": [
        {
          "name": "ClassRenderPartsDocsClassDescriptionMerge",
          "description": "class description",
          "superclass": {
            "name": "LitElement",
            "package": "lit"
          },
          "kind": "class",
          "tagName": "class-render-parts-docs-class-description-merge",
          "cssParts": [
            {
              "name": "part",
              "summary": "part summary",
              "description": "part description\n\nmultiline **markdown**"
            },
            {
              "name": "summary",
              "summary": "summary part summary",
              "description": "jsdoc part description"
            },
            {
              "name": "deprecated",
              "deprecated": true
            },
            {
              "name": "reason",
              "deprecated": "deprecation reason"
            }
          ],
          "customElement": true
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "class-render-parts-docs-class-description-merge",
          "declaration": {
            "name": "ClassRenderPartsDocsClassDescriptionMerge",
            "module": "src/class-render-parts-docs-class-description-merge.js"
          }
        }
      ]
    }
  ]
}